data_model/src/lib.rs

```diff
@@ -267,37 +272,134 @@ impl<EXPECTED, GOT> EnumTryAsError<EXPECTED, GOT> {
 #[cfg(feature = "std")]
 impl<EXPECTED: Debug, GOT: Debug> std::error::Error for EnumTryAsError<EXPECTED, GOT> {}
 
-/// Represents Iroha Configuration parameters.
+/// Identification of a [`Parameter`].
 #[derive(
     Debug,
     Display,
     Clone,
-    Copy,
+    FromStr,
     PartialEq,
     Eq,
     PartialOrd,
     Ord,
     Hash,
     Decode,
     Encode,
-    Deserialize,
-    Serialize,
+    DeserializeFromStr,
+    SerializeDisplay,
     FfiType,
     IntoSchema,
 )]
-pub enum Parameter {
-    /// Maximum amount of Faulty Peers in the system.
-    #[display(fmt = "Maximum number of faults is {_0}")]
-    MaximumFaultyPeersAmount(u32),
-    /// Maximum time for a leader to create a block.
-    #[display(fmt = "Block time: {_0}ms")]
-    BlockTime(u128),
-    /// Maximum time for a proxy tail to send commit message.
-    #[display(fmt = "Commit time: {_0}ms")]
-    CommitTime(u128),
-    /// Time to wait for a transaction Receipt.
-    #[display(fmt = "Transaction receipt time: {_0}ms")]
-    TransactionReceiptTime(u128),
+#[display(fmt = "{name}")]
+pub struct Id {
+    /// [`Name`] unique to a [`Parameter`].
+    pub name: Name,
+}
+
+declare_item! {
+    #[derive(
+        Debug, Display, Clone, IdOrdEqHash, Getters, Decode, Encode, DeserializeFromStr, SerializeDisplay, FfiType, IntoSchema,
+    )]
+    #[display(fmt = "?{id}={val}")]
+    #[getset(get = "pub")]
+    /// A chain-wide configuration parameter and its value.
+    pub struct Parameter {
+        /// Unique [`Id`] of the [`Parameter`].
+        id: Id,
+        /// Current value of the [`Parameter`].
+        val: Value,
+    }
+}
+
+impl FromStr for Parameter {
+    type Err = ParseError;
+
+    fn from_str(string: &str) -> Result<Self, Self::Err> {
+        if let Some((parameter_id_candidate, val_candidate)) = string.rsplit_once('=') {
+            if let Some(parameter_id_candidate) = parameter_id_candidate.strip_prefix('?') {
+                let param_id: <Parameter as Identifiable>::Id =
+                    parameter_id_candidate.parse().map_err(|_| ParseError {
+                        reason: "Failed to parse the `param_id` part of the `Parameter`.",
+                    })?;
+                if let Some((val, ty)) = val_candidate.rsplit_once('_') {
+                    let val = match ty {
+                        // Shorthand for `LengthLimits`
+                        "LL" => {
```

Comment:
I'd say it's mostly for the purpose of having a more compact representation in `genesis.json`. And I wouldn't exactly call them arbitrary, as these 3 types of limits have always been part of configuration and that is unlikely to change. Though I do agree that the only place for validation being `FromStr` is a bit weird, but that's the price to pay for looser and more flexible `Parameter` typing (as I met a dead end with a stronger typed version earlier).

Reply:
> having a more compact representation in `genesis.json`

I'm not sure it's worth it. I don't think editing `genesis.json` by hand is a frequent necessity.

> I wouldn't exactly call them arbitrary

By "arbitrary" I meant the parameter id itself, in the sense that it can be an arbitrary string of certain format _except_ for these carveouts for shorthand notation. I think such special-meaning values are a liability, and I'm not sure potential benefits here are worth it.

I don't insist though, since it's ultimately a minor detail. In case you strongly prefer having shorthand notation even if it's not necessarily something we _need_ to have (assuming I'm not wrong about `genesis.json`, of course), it's fine with me.

Reply:
> I don't think editing genesis.json by hand is a frequent necessity.

It's hard to know exactly what's going to be frequent right now, considering that our only user is Orillion pretty much, but judging from what we have in `kagami`, the only supported functionality is either a bootstrapped synthetic genesis mainly for testing purposes or a completely default one. So if you want something really custom, there's no other way but to do it by hand, and a lot of repetitiveness quickly gets unmanageable. So I'd argue for leaving it in currently. I'll open a reminder issue just in case though for further reconsideration of this matter.